pallets/fiat_filter/src/tests.rs

```diff
@@ -0,0 +1,762 @@
+            assert_noop!(
+                FiatFilterModule::root_set_update_freq(Origin::signed(ALICE), 42u64),
+                DispatchError::BadOrigin
+            );
+        });
+    }
+
+    #[test]
+    fn root_set_DockFiatRate__OK() {
+        ext().execute_with(|| {
+            // Dispatch a signed extrinsic.
+            let executed = FiatFilterModule::root_set_dock_fiat_rate(
+                RawOrigin::Root.into(),
+                Permill::from_parts(500_000),
+            );
+            assert_ok!(executed);
+
+            // Read pallet storage and assert an expected result.
+            assert_eq!(
+                FiatFilterModule::dock_fiat_rate(),
+                Permill::from_parts(500_000)
+            );
+        });
+    }
+
+    #[test]
+    fn root_set_dock_fiat_rate__Err_NotRoot() {
+        ext().execute_with(|| {
+            // Ensure the expected error is thrown when no value is present.
+            assert_noop!(
+                FiatFilterModule::root_set_update_freq(Origin::signed(ALICE), 42u64),
+                DispatchError::BadOrigin
+            );
+        });
+    }
+}
+
+mod tests_fail_modes {
+    use super::*;
+    use anchor;
+    use frame_support::dispatch::DispatchError;
+
+    #[test]
+    fn anchor_new__Err_no_balance() {
+        ext().execute_with(|| {
+            // empty alice's balance
+            let _ = <Test as Config>::Currency::make_free_balance_be(&ALICE, 0);
+            // prepare data
+            let dat = (0..32).map(|_| rand::random()).collect();
+            // execute call
+            let (_fee_microdock, executed) =
+                measure_fees(TestCall::AnchorMod(anchor::Call::<Test>::deploy(dat)));
+            assert_noop!(
+                executed,
+                DispatchError::Module {
+                    index: 0,
+                    error: 3,
+                    message: Some("InsufficientBalance")
+                },
+            );
+        });
+    }
+
+    #[test]
+    fn anchor_new__Err_insufficient_balance() {
+        ext().execute_with(|| {
+            // reduce alice's balance to just under the required fee
+            let _ = <Test as Config>::Currency::make_free_balance_be(&ALICE, 26653490);
+            // prepare data
+            let dat = (0..32).map(|_| rand::random()).collect();
+            // execute call
+            let (_fee_microdock, executed) =
+                measure_fees(TestCall::AnchorMod(anchor::Call::<Test>::deploy(dat)));
+            assert_noop!(
+                executed,
+                DispatchError::Module {
+                    index: 0,
+                    error: 3,
+                    message: Some("InsufficientBalance")
+                },
+            );
+        });
+    }
+
+    #[test]
+    fn anchor_new__Err_overflow() {
+        ext().execute_with(|| {
+            // set the dock_fiat_rate to the minimum to trigger overflow
+            let executed = FiatFilterModule::root_set_dock_fiat_rate(
```

Comment:
I see it very unlikely but if the price goes too low, the number of tokens should be high. We can set a reasonable minimum value in code but rejecting txns (due to overflow) when price drops doesn't seem like the right thing to do.

Reply:
Now with the fees expressed as u64 like the Runtime Balance and the rate in the same unit as the PriceProvider, I think there is no possible way to trigger this overflow anymore without going over the total DOCK supply.

Does it still make sense to test anything here ?